### PR TITLE
Add prefix `list-` to --interfaces and --reporters options

### LINKED ISSUE #3777: --interfaces and --reporters should be --list-interfaces and --list-reporters, respectively
1. `--reporters` would be nice to alias to `--reporter` if/when we support multiple reporters; so let's use `--list-reporters` instead
1. `--interfaces` should be `--list-interfaces` for consistency :smile:

I probably would *not* call this change semver-major, even if it might be per our usual interpretation.  @mochajs/core thoughts?

### PR DESCRIPTION
**Pull request by ParkSB**
### Description of the Change

* Rename `--interfaces` to `--list-interfaces`.
* Rename `--reporters` to `--list-reporters`.
* ~Add `--reporters` alias for `--reporter`.~

### Alternate Designs

N/A

### Benefits

It makes guessing a behavior of the options easier.

### Possible Drawbacks

N/A

### Applicable issues

Closes #3777

### CODE REVIEW

File: lib/cli/run-option-metadata.js [hunk 1]
```diff
@@ -76,7 +76,7 @@ exports.aliases = {
   ignore: ['exclude'],
   invert: ['i'],
   'no-colors': ['C'],
-  reporter: ['R'],
+  reporter: ['reporters', 'R'],
```
[the remark this answers is not in the document]
Comment (dasilvacontin) [reply lib/cli/run-option-metadata.js]: Afaik this option can only be specified once, and the value must be a `string` (not an `array`), therefore it doesn't make sense to have an alias with the plural of the word.

[the remark this answers is not in the document]
Comment (dasilvacontin) [reply lib/cli/run-option-metadata.js]: Relevant lines:
- [`validatePlugin(argv, 'reporter', Mocha.reporters);` @ lib/cli/run.js](https://github.com/mochajs/mocha/blob/5d4dd98747637d0e7ed3007328ec9627dd7eda41/lib/cli/run.js#L278)
- [`throw new TypeError("--${key} <${key}>" can only be specified once);` @ validatePlugin, lib/cli/run-helpers.js](https://github.com/mochajs/mocha/blob/5d4dd98747637d0e7ed3007328ec9627dd7eda41/lib/cli/run-helpers.js#L159)

[the remark this answers is not in the document]
Comment (ParkSB) [reply lib/cli/run-option-metadata.js]: I thought we'll support multiple reporters soon. If not, `reporters` shouldn't be an alias for `reporter` now of course.

**Comment by outsideris** [comment]
[Changing document](https://github.com/mochajs/mocha/blob/master/docs/index.md#--reporter-name--r-name) should be included into this PR.

**Pull request merged**